src/boundaries.rs

```diff
@@ -102,7 +102,55 @@ pub fn build_boundary(
     relation: &osmpbfreader::Relation,
     objects: &BTreeMap<osmpbfreader::OsmId, osmpbfreader::OsmObj>,
 ) -> Option<MultiPolygon<f64>> {
-    let roles = ["outer", "enclave", ""];
+    use geo::algorithm::contains::Contains;
+
+    let outer_polys = build_boundary_parts(relation, objects, vec!["outer", "enclave", ""]);
+    let inner_polys = build_boundary_parts(relation, objects, vec!["inner"]);
+    let mut multipoly = MultiPolygon(vec![]);
+
+    if inner_polys.is_none() {
+        return outer_polys;
+    }
+    let mut inner_polys = inner_polys.unwrap();
+
+    if outer_polys.is_some() {
```

Comment:
I think you can rewrite this (not tested :stuck_out_tongue_winking_eye: ):

```rust
outer_polys.map(|outers|
    outers.map(|outer| {
        let mut i = 0;
        let mut inners = vec![];
        while(blablabla) {
            ...
        }
       Polygon::new(outer.exterior.clone(), inners)
    }).collect()
)

//this will already be an Option<Multipolygon>, you can return it directly
```